src/linker/Linker/LinkContext.cs

```diff
@@ -532,53 +532,53 @@ public void LogDiagnostic (string message)
 
 		/// <summary>
 		/// Display a warning message to the end user.
+		/// This API is used for warnings defined in the linker, not by custom steps. Warning
+		/// versions are inferred from the code, and every warning that we define is versioned.
 		/// </summary>
 		/// <param name="text">Humanly readable message describing the warning</param>
 		/// <param name="code">Unique warning ID. Please see https://github.com/mono/linker/blob/master/doc/error-codes.md for the list of warnings and possibly add a new one</param>
 		/// <param name="origin">Filename or member where the warning is coming from</param>
 		/// <param name="subcategory">Optionally, further categorize this warning</param>
-		/// <param name="version">Optional warning version number. Versioned warnings can be controlled with the
-		/// warning wave option --warn VERSION. Unversioned warnings are unaffected by this option. </param>
 		/// <returns>New MessageContainer of 'Warning' category</returns>
-		public void LogWarning (string text, int code, MessageOrigin origin, string subcategory = MessageSubCategory.None, WarnVersion? version = null)
+		public void LogWarning (string text, int code, MessageOrigin origin, string subcategory = MessageSubCategory.None)
```

Comment:
I thought we would keep the optional parameter for custom steps - as for those we can't really infer the version.

Reply:
My thinking is that we have CreateWarningMessage, exposed in the ref assembly, as the API for custom steps, taking an optional warning version. The LinkContext.LogWarning helpers are not exposed in the ref assembly and would be refactored to just take a code - does that seem reasonable?

Reply:
Didn't know we don't expose these in ref assembly - in that case I completely agree.

Reply:
Great - I can see some of those helpers being useful for custom steps too - I figure we can agree on additional APIs later.